### ISSUE TITLE
Adding RNLoadingButton to Interface Builder crashes auto layout agent

### ISSUE THREAD

**Issue by DmitryMoroz** [souzainf3/RNLoadingButton-Swift#17]
At Xcode 8 (8.3.3) using this control causes IB agent crash on render with error: "failed to render and update auto layout status agent crashes". This error doesn't affect compiling the project, but disable work at IB with screen.
I took the log from system:

> *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'An instance 0x7f816f67d240 of class RNLoadingButton_Swift.RNLoadingButton was deallocated while key value observers were still registered with it. Current observation info: <NSKeyValueObservationInfo 0x7f8170bd0fa0> (
> <NSKeyValueObservance 0x7f8170bd1530: Observer: 0x7f816f67d240, Key path: self.state, Options: <New: YES, Old: NO, Prior: NO> Context: 0x0, Property: 0x7f816f57e8d0>
> <NSKeyValueObservance 0x7f8170bd1960: Observer: 0x7f8170bd1530, Key path: state, Options: <New: YES, Old: NO, Prior: NO> Context: 0x0, Property: 0x7f816f6471a0>
> <NSKeyValueObservance 0x7f8170bd1db0: Observer: 0x7f816f67d240, Key path: self.selected, Options: <New: YES, Old: NO, Prior: NO> Context: 0x0, Property: 0x7f816f64aa90>
> <NSKeyValueObservance 0x7f8170bd1cc0: Observer: 0x7f8170bd1db0, Key path: selected, Options: <New: YES, Old: NO, Prior: NO> Context: 0x0, Property: 0x7f816f645cc0>
> <NSKeyValueObservance 0x7f8170bd2160: Observer: 0x7f816f67d240, Key path: self.highlighted, Options: <New: YES, Old: NO, Prior: NO> Context: 0x0, Property: 0x7f816f63db60>
> <NSKeyValueObservance 0x7f8170bd1b70: Observer: 0x7f8170bd2160, Key path: highlighted, Options: <New: YES, Old: NO, Prior: NO> Context: 0x0, Property: 0x7f816f64b6d0>
> )'
> terminating with uncaught exception of type NSException

Found that you make init for IB:

```
override open func prepareForInterfaceBuilder() {
        super.prepareForInterfaceBuilder()
        
        #if !TARGET_INTERFACE_BUILDER
            // this code will run in the app itself
        #else
            // this code will execute only in IB
            self.setupActivityIndicator()
            commonInit()
        #endif
    }
```

But deinit function with remove observers hidden from IB with macros:

 ```
#if !TARGET_INTERFACE_BUILDER

    deinit {
        self.removeObservers()
    }

#endif
```

I fixed it by removing observers for IB:
```

fileprivate func commonInit() {
        self.adjustsImageWhenHighlighted = true
        self.storeDefaultValues()
        #if !TARGET_INTERFACE_BUILDER
            self.addObservers()
        #endif
    }
```

**Comment by souzainf3** [comment]
Thanks for your contribuitions.

I will happy with an pull request fixing it.

**Comment by souzainf3** [comment]
Fixed on version 5.0.0. Thanks @DmitryMoroz
